Objects/listobject.c

```diff
@@ -31,6 +31,48 @@ get_list_freelist(void)
 }
 #endif
 
+static PyListObject *
+list_new_prealloc(Py_ssize_t size)
+{
+    PyListObject *op;
+    assert(size >= 0);
+#ifdef WITH_FREELISTS
+    struct _Py_list_freelist *list_freelist = get_list_freelist();
+    if (PyList_MAXFREELIST && list_freelist->numfree > 0) {
+        list_freelist->numfree--;
+        op = list_freelist->items[list_freelist->numfree];
+        OBJECT_STAT_INC(from_freelist);
+        _Py_NewReference((PyObject *)op);
+    }
+    else
+#endif
+    {
+        op = PyObject_GC_New(PyListObject, &PyList_Type);
+        if (op == NULL) {
+            return NULL;
+        }
+    }
+    if (size <= 0) {
+        op->ob_item = NULL;
+        op->allocated = 0;
+    }
+    else {
+        size_t capacity = size;
+        PyObject **items = (PyObject **) PyMem_Calloc(size, sizeof(PyObject *));
```

Comment:
The previous code did not zero initialize `ob_item`. The callers to `list_new_prealloc` were responsible for initializing ob_item.